diff --color -Naur a/src/generator.c b/src/generator.c
--- a/src/generator.c	2022-02-02 15:19:36.350038791 +0100
+++ b/src/generator.c	2022-02-02 15:20:04.809161098 +0100
@@ -626,7 +626,7 @@
   quad_unit_file_add (service, UNIT_GROUP,
                       "RequiresMountsFor", "%t/containers");
 
-  g_autofree char *exec_cond = g_strdup_printf ("/usr/bin/bash -c \"! /usr/bin/podman volume exists %s\"", volume_name);
+  g_autofree char *exec_cond = g_strdup_printf ("/bin/sh -c \"! /usr/bin/podman volume exists %s\"", volume_name);
 
   g_auto(GStrv) labels = quad_unit_file_lookup_all (container, VOLUME_GROUP, "Label");
   g_autoptr(GHashTable) podman_labels = parse_keys (labels);
