--- htop-0.8.3.origin/configure.ac	2009-06-23 15:53:18.000000000 +0200
+++ htop-0.8.3/configure.ac	2010-05-07 14:59:26.130538859 +0200
@@ -69,6 +69,11 @@
    AC_DEFINE(HAVE_OPENVZ, 1, [Define if openvz support enabled.])
 fi
 
+AC_ARG_ENABLE(cgroup, [AC_HELP_STRING([--enable-cgroup], [enable cgroups support])], ,enable_group="no")
+if test "x$enable_cgroup" = xyes; then
+   AC_DEFINE(HAVE_CGROUP, 1, [Define if cgroup support enabled.])
+fi
+
 AC_ARG_ENABLE(vserver, [AC_HELP_STRING([--enable-vserver], [enable VServer support])], ,enable_vserver="no")
 if test "x$enable_vserver" = xyes; then
     AC_DEFINE(HAVE_VSERVER, 1, [Define if vserver support enabled.])
--- htop-0.8.3.origin/Process.c	2009-06-02 21:17:57.000000000 +0200
+++ htop-0.8.3/Process.c	2010-05-07 12:10:31.458635669 +0200
@@ -52,6 +52,9 @@
    #ifdef HAVE_OPENVZ
    CTID, VPID,
    #endif
+   #ifdef HAVE_CGROUP
+   CGROUP,
+   #endif
    #ifdef HAVE_VSERVER
    VXID,
    #endif
@@ -130,6 +133,9 @@
    unsigned int ctid;
    unsigned int vpid;
    #endif
+   #ifdef HAVE_CGROUP
+   char* cgroup;
+   #endif
    #ifdef HAVE_VSERVER
    unsigned int vxid;
    #endif
@@ -168,6 +174,9 @@
 #ifdef HAVE_OPENVZ
    "CTID", "VPID",
 #endif
+#ifdef HAVE_CGROUP
+   "CGROUP",
+#endif
 #ifdef HAVE_VSERVER
    "VXID",
 #endif
@@ -190,6 +199,9 @@
 #ifdef HAVE_OPENVZ
    " CTID ", " VPID ",
 #endif
+#ifdef HAVE_CGROUP
+   "    CGROUP ",
+#endif
 #ifdef HAVE_VSERVER
    " VXID ",
 #endif
@@ -403,6 +415,9 @@
    case CTID: snprintf(buffer, n, "%5u ", this->ctid); break;
    case VPID: snprintf(buffer, n, "%5u ", this->vpid); break;
    #endif
+   #ifdef HAVE_CGROUP
+   case CGROUP: snprintf(buffer, n, "%-10s ", this->cgroup); break;
+   #endif
    #ifdef HAVE_VSERVER
    case VXID: snprintf(buffer, n, "%5u ", this->vxid); break;
    #endif
@@ -580,6 +595,10 @@
    case VPID:
       return (p1->vpid - p2->vpid);
    #endif
+   #ifdef HAVE_CGROUP
+   case CGROUP:
+      return strcmp(p1->cgroup ? p1->cgroup : "", p2->cgroup ? p2->cgroup : "");
+   #endif
    #ifdef HAVE_VSERVER
    case VXID:
       return (p1->vxid - p2->vxid);
--- htop-0.8.3.origin/ProcessList.c	2009-03-11 14:03:42.000000000 +0100
+++ htop-0.8.3/ProcessList.c	2010-05-10 14:20:42.381533877 +0200
@@ -640,6 +640,26 @@
             }
             #endif
 
+            #ifdef HAVE_CGROUP
+	    if (access("/proc/cgroups", R_OK) != 0) {
+               process->cgroup = strdup("       ");
+            } else {
+	       snprintf(statusfilename, MAX_NAME, "%s/%s/cgroup", dirname, name);
+	       status = ProcessList_fopen(this, statusfilename, "r");
+	       if (status == NULL)
+                  goto errorReadingProcess;
+               char cg_tmpchar[256];
+               fgets(cg_tmpchar,255,status);
+	       char* cg_trimed = String_trim(cg_tmpchar);
+               char** cg_fields = String_split(cg_trimed, ':');
+	       free(cg_trimed);
+	       char* cg_value = String_cat(cg_fields[2],"       ");
+	       String_freeArray(cg_fields);
+               process->cgroup=strndup(cg_value+1,10);
+	       free(cg_value);
+               fclose(status);
+	    };
+	    #endif
             #ifdef HAVE_VSERVER
             snprintf(statusfilename, MAX_NAME, "%s/%s/status", dirname, name);
             status = ProcessList_fopen(this, statusfilename, "r");
