diff -Naur synergy-1.4.14-Source.old/src/gui/src/MainWindow.cpp synergy-1.4.14-Source/src/gui/src/MainWindow.cpp
--- synergy-1.4.14-Source.old/src/gui/src/MainWindow.cpp	2013-10-16 19:20:32.000000000 +0200
+++ synergy-1.4.14-Source/src/gui/src/MainWindow.cpp	2013-10-22 08:53:05.907038255 +0200
@@ -401,10 +401,12 @@
 #endif
 	}
 
+#if !defined(Q_OS_LINUX)
 	if (appConfig().isPremium())
 	{
 		args << "--enable-drag-drop";
 	}
+#endif
 
 	if ((synergyType() == synergyClient && !clientArgs(args, app))
 		|| (synergyType() == synergyServer && !serverArgs(args, app)))
