diff -Naur /root/gpa-0.9.4.old/src/cm-openpgp.c /root/gpa-0.9.4/src/cm-openpgp.c
--- /root/gpa-0.9.4.old/src/cm-openpgp.c	2012-08-16 11:26:42.000000000 +0200
+++ /root/gpa-0.9.4/src/cm-openpgp.c	2013-10-25 18:15:27.052150034 +0200
@@ -136,6 +136,8 @@
     case 0x0005: return "ZeitControl";
 
     case 0x002A: return "Magrathea";
+
+    case 0x7975: return "Yubico";
       /* 0x00000 and 0xFFFF are defined as test cards per spec,
          0xFFF00 to 0xFFFE are assigned for use with randomly created
          serial numbers.  */
