Description: Fix FTBFS with gcc 4.7
Author: Cyril Brulebois <kibi@debian.org>
Bug-Debian: http://bugs.debian.org/667428
--- zoneminder-1.25.0.orig/src/zm_thread.h
+++ zoneminder-1.25.0/src/zm_thread.h
@@ -21,6 +21,7 @@
 #define ZM_THREAD_H
 
 #include <pthread.h>
+#include <unistd.h>
 #include "zm_exception.h"
 #include "zm_utils.h"
 
--- zoneminder-1.25.0.orig/src/zm_logger.cpp
+++ zoneminder-1.25.0/src/zm_logger.cpp
@@ -33,6 +33,7 @@
 #include <signal.h>
 #include <stdarg.h>
 #include <errno.h>
+#include <unistd.h>
 
 bool Logger::smInitialised = false;
 Logger *Logger::smInstance = 0;
